src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_ARM64/ARM64ReadyToRunHelperNode.cs

```diff
@@ -231,7 +231,76 @@ protected override void EmitCode(NodeFactory factory, ref ARM64Emitter encoder,
         // may trash volatile registers. (there are calls to the slow helper and possibly to the platform's TLS support)
         private static void EmitInlineTLSAccess(NodeFactory factory, ref ARM64Emitter encoder)
         {
-            throw new NotImplementedException();
+            ISymbolNode getInlinedThreadStaticBaseSlow = factory.HelperEntrypoint(HelperEntrypoint.GetInlinedThreadStaticBaseSlow);
+            ISymbolNode tlsRoot = factory.TlsRoot;
+            // IsSingleFileCompilation is not enough to guarantee that we can use "Initial Executable" optimizations.
+            // we need a special compiler flag analogous to /GA. Just assume "false" for now.
+            // bool singleFileExe = factory.CompilationModuleGroup.IsSingleFileCompilation;
+            bool singleFileExe = false;
+
+            if (factory.Target.OperatingSystem == TargetOS.Linux)
+            {
+                if (singleFileExe)
```

Comment:
the difference between LE and TLSDESC variants is  `3+5` vs. `6+5` instructions, not counting the frame setup and ret, as that would not be needed in JIT case.

That would be 24 vs. 44 bytes. It makes support for Initial Exec compiler switch a lot more compelling.

Reply:
The tracking issue for the "optimize for application" switch: https://github.com/dotnet/runtime/issues/87642